Cargo.toml

```diff
@@ -258,29 +258,32 @@ rusty-fork = "0.3.0"
+default-no-api-client = ["api", "sources", "vrl-cli", "transforms", "sinks", "vendor-all", "unix", "alloc-jemalloc", "leveldb", "rdkafka-plain"]
+default-no-vrl-cli = ["api", "sources", "transforms", "sinks", "vendor-all", "unix", "alloc-jemalloc", "leveldb", "rdkafka-plain"]
 
 # Target specific release features.
 # The `make` tasks will select this according to the appropriate triple.
 # Use this section to turn off or on specific features for specific triples.
-target-x86_64-unknown-linux-gnu = ["api", "api-client", "sources", "transforms", "sinks", "vendor-all", "unix", "leveldb", "rdkafka-cmake"]
-target-aarch64-unknown-linux-gnu = ["api", "api-client", "sources", "transforms", "sinks", "vendor-openssl", "vendor-libz", "unix", "leveldb", "rdkafka-cmake"]
-target-x86_64-unknown-linux-musl = ["api", "api-client", "sources", "transforms", "sinks", "vendor-openssl", "vendor-libz", "unix",  "leveldb", "rdkafka-cmake"]
-target-aarch64-unknown-linux-musl = ["api", "api-client", "sources", "transforms", "sinks", "vendor-openssl", "vendor-libz", "unix",  "leveldb", "rdkafka-cmake"]
-# TODO: Enable leveldb here for armv7-unknown-linux-musleabihf
-target-armv7-unknown-linux-musleabihf = ["api", "api-client", "sources", "transforms", "sinks", "vendor-openssl", "vendor-libz", "rdkafka-cmake"]
-target-armv7-unknown-linux-gnueabihf = ["api", "api-client", "sources", "transforms", "sinks", "vendor-openssl", "vendor-libz", "unix", "leveldb", "rdkafka-cmake"]
+target-x86_64-unknown-linux-gnu = ["api", "api-client", "sources", "transforms", "sinks", "vendor-all", "unix", "alloc-jemalloc", "leveldb", "rdkafka-cmake"]
+target-aarch64-unknown-linux-gnu = ["api", "api-client", "sources", "transforms", "sinks", "vendor-openssl", "vendor-libz", "unix", "alloc-jemalloc", "leveldb", "rdkafka-cmake"]
+target-x86_64-unknown-linux-musl = ["api", "api-client", "sources", "transforms", "sinks", "vendor-openssl", "vendor-libz", "unix", "alloc-jemalloc", "leveldb", "rdkafka-cmake"]
+target-aarch64-unknown-linux-musl = ["api", "api-client", "sources", "transforms", "sinks", "vendor-openssl", "vendor-libz", "unix", "alloc-jemalloc", "leveldb", "rdkafka-cmake"]
+# TODO: Enable leveldb and alloc-jemalloc  here for armv7-unknown-linux-musleabihf
+target-armv7-unknown-linux-musleabihf = ["api", "api-client", "sources", "transforms", "sinks", "vendor-openssl", "vendor-libz", "alloc-system", "rdkafka-cmake"]
+target-armv7-unknown-linux-gnueabihf = ["api", "api-client", "sources", "transforms", "sinks", "vendor-openssl", "vendor-libz", "unix", "alloc-jemalloc", "leveldb", "rdkafka-cmake"]
 
 # Enables features that work only on systems providing `cfg(unix)`
-unix = ["jemallocator"]
+unix = []
+# Allocator choices.
+alloc-jemalloc = ["jemallocator"] # use jemalloc
+alloc-system = [] # noop, just to explicitly state that we want to use system allocator
```

Comment:
I'm not sure that I'm in favor of this. I get that it could be seen as explicitness, but it doesn't actually do anything and that seems more confusing than just knowing that the system allocator is the default (true for every application).

Reply:
Without it, it's not clear which ones of our feature definitions are supposed to set the allocator. I.e. it's a hint to where you'd set the jemalloc allocator if you need to swap the system one to it.

Our "system of futures" in `Cargo.toml` has grown pretty big over time, and it was not easy to get a grasp on it. However, I don't think it can be simplified sanely in the current state of `cargo`, so adding a more logical structure to the features is the only improvement I think we can aim for.

So, the confusion should be solved by the comment, but the hints/placeholders remain. You won't overlook/loose them if you're moving/changing the features (the alleged cause of why the benches ended up not using jemalloc).